logstash-core/lib/logstash/plugin.rb

```diff
@@ -92,7 +92,7 @@ def debug_info
   end
 
   def metric=(new_metric)
-    @metric = new_metric.namespace(@id)
+    @metric = new_metric
```

Comment:
Thanks for noticing it.

I have struggled with this change a bit, when I changed for a `NamespacedMetric` I though it was more natural to provide a context to the plugin. Since the plugins are actually instantiated in the delegators and can be run in multiples thread I though It was better generate the metric context there to make it more granular.

As a plugin developer he doesnt need to know what logic need to be done to have his keys to be correctly routed to the backend. 

Do you see any advantages to make it there? I am happy to discuss that!